rq/registry.py

```diff
@@ -270,6 +270,7 @@ def cleanup(self, timestamp: Optional[float] = None):
                         job.save(pipeline=pipeline, include_meta=False)
                         job.cleanup(ttl=-1, pipeline=pipeline)
                         failed_job_registry.add(job, job.failure_ttl)
+                        queue.enqueue_dependents(job)
```

Comment:
You'll need to check whether the parent job has `allow_failure=True`. Please add a test case for this as well.

Reply:
@selwin, I'm not sure why I need to check whether the parent job of this `job` has `allow_dependency_failures == True` or not.

Let's imagine a situation when there are 3 jobs in such sequence:
J1  <- J2 (`allow_dependency_failures=False`) <- J3 (`allow_dependency_failures=True`)

If `J1` is finished and `J2` is started, then after docker container stops, `J2` should be moved to `FailedJobRegistry`, but `J3` should be added to the queue (despite that `J2` as a parent job has `allow_dependency_failures==False`
If `J1` fails, then `J2` will not be started at all.
In addition, If a job has been started previously (because it is in the StartedJobRegistry), that means it has already passed the check on whether it can run or not, based on the parent job.

This PR does not fix the case when `J1` is started, the docker container stops, and `J1` is moved to the `FailedJobRegistry` while `J2` and `J3` remain in the `DeferredJobRegistry` because such a bug also occurs when the `StartedJobRegistry.cleanup` method is not called.

For instance:
```python
job1 = queue.enqueue_call(div_by_zero)
job2 = queue.enqueue_call(say_hello, depends_on=Dependency(jobs=job1, allow_failure=False))
job3 = queue.enqueue_call(say_hello, depends_on=Dependency(jobs=job2, allow_failure=True))
worker.work(burst=True)
job1.is_failed # True
job2.is_deferred # True
job3.is_deferred # True
```
I guess that in the example mentioned above, `J1` and `J2` should finally be moved to `FailedJobRegistry`, but `J3` should be added to `FinishedJobRegistry`.

Reply:
You're right.